datafusion/src/physical_plan/math_expressions.rs

```diff
@@ -116,3 +116,21 @@ math_unary_function!("exp", exp);
 math_unary_function!("ln", ln);
 math_unary_function!("log2", log2);
 math_unary_function!("log10", log10);
+
+/// random SQL function
+pub fn random(args: &[ColumnarValue]) -> Result<ColumnarValue> {
+    let len = match &args[0] {
+        ColumnarValue::Array(array) => array.len(),
+        _ => {
+            return Err(DataFusionError::Internal(
+                "Expect random function to take no param".to_string(),
+            ))
+        }
+    };
+    let mut rng = thread_rng();
+    let mut array = Vec::with_capacity(len);
+    for _ in 0..len {
+        array.push(Some(rng.gen_range(0.0..1.0)))
```

Comment:
`arrow::compute::kernels::arity::unary` could be probably used here.

see also:
https://github.com/apache/arrow-datafusion/pull/309

Reply:
not yet, i guess https://github.com/apache/arrow-rs/pull/275 is needed first?

Reply:
Yes, makes sense, thanks!

Reply:
`PrimitiveArray::<f64>::from_iter_values` would work here.

Reply:
Updated per suggestions